src/realm/exceptions.hpp

```diff
@@ -355,6 +355,19 @@ class FileAccessError : public RuntimeError {
 };
 
 struct SystemError : RuntimeError {
+    struct ExtraInfo : public Status::ExtraInfo {
+        explicit ExtraInfo(std::error_code ec)
+            : ec(ec)
+        {
+        }
+        std::error_code ec;
+    };
+
+    ~SystemError() noexcept override;
+
+    static Status make_status(std::error_code err, std::string_view msg, bool msg_is_prefix = false);
+    static std::error_code get_system_error_from_status(Status status);
```

Comment:
I haven't checked your std::error_code wip diff; will this only function be used to retrieve the error_code for SystemErrors? Or will this function go away at some point?

Reply:
I think there's actually value in letting SystemError reference a std::error_code and wasn't planning on removing these as part of my work - the main thing I wanted to do was extract std::error_code from inside Status and make it a part of SystemError only.